app/frontend/actions.js

```diff
@@ -66,33 +65,30 @@ module.exports = ({setState, getState}) => {
 
   const loadWallet = async (state, {cryptoProvider, secret}) => {
     loadingAction(state, 'Loading wallet data...', {walletLoadingError: undefined})
-    switch (cryptoProvider) {
-      case 'trezor':
-        wallet = await Cardano.CardanoWallet({
-          cryptoProvider: 'trezor',
-          config: ADALITE_CONFIG,
-          network: 'mainnet',
-          derivationScheme: derivationSchemes.v2,
-        })
-        break
-      case 'mnemonic':
-        secret = secret.trim()
-        wallet = await Cardano.CardanoWallet({
-          cryptoProvider: 'mnemonic',
-          mnemonicOrHdNodeString: secret,
-          config: ADALITE_CONFIG,
-          network: 'mainnet',
-          derivationScheme: derivationSchemes.v1,
-        })
-        break
-      default:
-        return setState({
-          loading: false,
-          walletLoadingError: {
-            code: 'UnknownCryptoProvider',
-            params: {cryptoProvider},
-          },
-        })
+    try {
+      wallet = await CryptoProviderFactory.getWallet(cryptoProvider, secret)
+    } catch (e) {
+      debugLog(e)
+      const code = e.name === 'UnknownCryptoProviderError' ? 'UnknownCryptoProviderError' : 'WalletInitializationError'
```

Comment:
Shouldn't this be set in following switch as well? Seems a bit ad-hoc to have this logic here.